cli/src/cmd_net.rs

```diff
@@ -0,0 +1,1475 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+// Copyright 2024 Oxide Computer Company
+
+use std::{collections::HashMap, net::IpAddr};
+
+use crate::RunnableCmd;
+use anyhow::Result;
+use async_trait::async_trait;
+use clap::Parser;
+use colored::*;
+use futures::TryStreamExt;
+use oxide::{
+    context::Context,
+    types::{
+        Address, AddressConfig, BgpAnnounceSetCreate, BgpAnnouncementCreate, BgpPeer,
+        BgpPeerConfig, BgpPeerStatus, ImportExportPolicy, IpNet, LinkConfigCreate, LinkFec,
+        LinkSpeed, LldpServiceConfigCreate, Name, NameOrId, Route, RouteConfig,
+        SwitchInterfaceConfigCreate, SwitchInterfaceKind, SwitchInterfaceKind2, SwitchLocation,
+        SwitchPort, SwitchPortConfigCreate, SwitchPortGeometry, SwitchPortGeometry2,
+        SwitchPortSettingsCreate,
+    },
+    Client, ClientSystemHardwareExt, ClientSystemNetworkingExt,
+};
+use serde::{Deserialize, Serialize};
+use std::io::Write;
+use tabwriter::TabWriter;
+use uuid::Uuid;
+
+// We do not yet support port breakouts, but the API is phrased in terms of
+// ports that can be broken out. The constant phy0 represents the first port
+// in a breakout.
+const PHY0: &str = "phy0";
+
+#[derive(Parser, Debug, Clone)]
+#[command(verbatim_doc_comment)]
+#[command(name = "port")]
+pub struct CmdPort {
+    #[clap(subcommand)]
+    subcmd: PortSubCommand,
+}
+
+#[async_trait]
+impl RunnableCmd for CmdPort {
+    async fn run(&self, ctx: &Context) -> Result<()> {
+        match &self.subcmd {
+            PortSubCommand::Config(cmd) => cmd.run(ctx).await,
+            PortSubCommand::Status(cmd) => cmd.run(ctx).await,
+        }
+    }
+}
+
+#[derive(Parser, Debug, Clone)]
+enum PortSubCommand {
+    /// Manage port configuration.
+    Config(CmdPortConfig),
+
+    /// Observe port status.
+    Status(CmdPortStatus),
+}
+
+#[derive(Parser, Debug, Clone)]
```

Comment:
I don't claim to really understand what this does, but perhaps a short doc comment explaining very briefly the command and (probably) that the subcommands are short-hand for common operations one could perform via `system networking switch-port-settings create`; if it's easy enough, I think a short doc comment on all the new commands and their subcommands would be great.